File: src/main/java/gq/noxiuam/bitches/object/type/WaifuBitch.java
package gq.noxiuam.bitches.object.type;

import gq.noxiuam.bitches.object.Bitch;
import lombok.Getter;

import java.util.*;

@Getter
public class WaifuBitch extends Bitch {

    private final boolean wifeMaterial;
    // every man has at least one of these for his waifu.
    private final List<String> sexualNicknames;
    private final String favoriteAnime;

    public WaifuBitch(boolean wifeMaterial, List<String> sexualNicknames, String favoriteAnime) {
        this.wifeMaterial = wifeMaterial;
        this.sexualNicknames = sexualNicknames;
        this.favoriteAnime = favoriteAnime;
        this.build();
    }

}

File: src/main/kotlin/gq/noxiuam/bitches/object/type/WaifuBitch.kt
package gq.noxiuam.bitches.`object`.type

import gq.noxiuam.bitches.`object`.Bitch
import lombok.Getter

@Getter
class WaifuBitch(
    private val wifeMaterial: Boolean, // every man has at least one of these for his waifu.
    private val sexualNicknames: List<String>, private val favoriteAnime: String
) : Bitch() {
    init {
        this.build()
    }
}